src/devtools/client/debugger/src/components/SecondaryPanes/Frames/NewFrames.tsx

```diff
@@ -0,0 +1,147 @@
+import { Suspense, useMemo } from "react";
+import { PauseId } from "@replayio/protocol";
+import { ThreadFront } from "protocol/thread";
+import { useAppDispatch, useAppSelector } from "ui/setup/hooks";
+import {
+  getFrameworkGroupingState,
+  getPauseId,
+  getSelectedFrame,
+  getThreadContext,
+} from "devtools/client/debugger/src/selectors";
+import { isCurrentTimeInLoadedRegion } from "ui/reducers/app";
+import { getSourceDetailsEntities, getSourcesLoading } from "ui/reducers/sources";
+import { formatCallStackFrames } from "../../../selectors/getCallStackFrames";
+import { selectFrame as selectFrameAction } from "../../../actions/pause/selectFrame";
+import { toggleFrameworkGrouping as setFrameworkGroupingAction } from "../../../reducers/ui";
+import { enterFocusMode as enterFocusModeAction } from "ui/actions/timeline";
+import { getAsyncParentFramesSuspense } from "ui/suspense/util";
+import { createFrame } from "../../../client/create";
+import { collapseFrames, formatCopyName } from "../../../utils/pause/frames";
+import { copyToTheClipboard } from "../../../utils/clipboard";
+import ErrorBoundary from "bvaughn-architecture-demo/components/ErrorBoundary";
+import Frame from "./Frame";
+import Group from "./Group";
+import { CommonFrameComponentProps } from ".";
+
+function FramesRenderer({
+  panel,
+  pauseId,
+  asyncIndex = 0,
+}: {
+  panel: "debugger" | "console" | "webconsole";
+  pauseId: PauseId;
+  asyncIndex?: number;
+}) {
+  const state = useAppSelector(state => state);
+  const cx = getThreadContext(state);
+  const sources = getSourceDetailsEntities(state);
+  const frameworkGroupingOn = getFrameworkGroupingState(state);
+  const selectedFrame = getSelectedFrame(state);
+  const dispatch = useAppDispatch();
+
+  function toggleFrameworkGrouping() {
+    dispatch(setFrameworkGroupingAction(!frameworkGroupingOn));
+  }
+  function selectFrame(...args: Parameters<typeof selectFrameAction>) {
+    dispatch(selectFrameAction(...args));
+  }
+  //TODO copy async parent frames as well
```

Comment:
Do we have a Linear issue filed for this follow up? (I'm not sure what it entails TBH.)

Reply:
Just created FE-750